urlshort/handler.go

```diff
@@ -32,6 +46,32 @@ func MapHandler(pathsToUrls map[string]string, fallback http.Handler) http.Handl
 // See MapHandler to create a similar http.HandlerFunc via
 // a mapping of paths to urls.
 func YAMLHandler(yml []byte, fallback http.Handler) (http.HandlerFunc, error) {
-	// TODO: Implement this...
-	return nil, nil
+	// Parse yml to a map
+	pathsToUrls := make(map[string]string)
+	seq := make([]Item, 0)
+	err := yaml.Unmarshal(yml, &seq)
+	if err != nil {
+		return nil, err
+	}
+	log.Printf("Seq %+v \n", seq)
+	for _, item := range seq {
+		pathsToUrls[item.Path] = item.Url
+	}
+	log.Printf("pathsToUrls %v \n", pathsToUrls)
+	mapHandler := MapHandler(pathsToUrls, fallback)
+	return mapHandler, nil
+}
+
+/* Example YAML:
+`
+- path: /urlshort
+  url: https://github.com/gophercises/urlshort
+- path: /urlshort-final
+  url: https://github.com/gophercises/urlshort/tree/solution
+`
+*/
+
+type Item struct {
```

Comment:
This name is too generic, especially since it is not in a package with a clarifying name.

Reply:
OK